app/src/main/java/com/kickstarter/services/KSApolloClient.kt

```diff
@@ -271,6 +268,37 @@ class KSApolloClient(val service: ApolloClient) : ApolloClientType {
         }.subscribeOn(Schedulers.io())
     }
 
+    override fun createComment(comment: PostCommentInput): Observable<Comment> {
+        return Observable.defer {
+            val ps = PublishSubject.create<Comment>()
+            this.service.mutate(
+                CreateCommentMutation.builder()
+                    .parentId(comment.parentId)
+                    .commentableId(comment.commentableId() ?: "")
+                    .clientMutationId(comment.clientMutationId)
+                    .body(comment.body ?: "")
+                    .build()
+            )
+                .enqueue(object : ApolloCall.Callback<CreateCommentMutation.Data>() {
+                    override fun onFailure(exception: ApolloException) {
+                        ps.onError(exception)
+                    }
+
+                    override fun onResponse(response: Response<CreateCommentMutation.Data>) {
+                        if (response.hasErrors()) {
+                            ps.onError(java.lang.Exception(response.errors?.first()?.message))
+                        }
+
+                        ps.onNext(
+                            createCommentObject(response.data?.createComment()?.comment()?.fragments()?.comment())
```

Comment:
Why do we need to create a comment from the mutation response? 
- Unless specified otherwise,  when using the post comment mutation we just need to check no error is returned from in that operation.
Take a look to other mutations as `com.kickstarter.services.KSApolloClient#updateUserEmail` or `com.kickstarter.services.KSApolloClient#updateUserPassword`

Reply:
I felt it will make sense to have a copy of what you posted. just in case we want to update the list without doing a full refresh

Reply:
Perfect, it makes sense for now! then please add some documentation on the method `createCommentObject` explaining your reasoning, something like you just explained.
`make a copy of what you posted. just in case we want to update the list without doing a full refresh` so we can know in the future the reasoning behind it, to keep it or deleted in the future in case no necessary